libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -98,10 +99,18 @@ internal class RouteRefreshController(
         }
     }
 
-    private val routeRefreshCallback = object : RouteRefreshCallback {
+    private fun createRouteRefreshCallback(
+        oldDirectionsRoute: DirectionsRoute,
+        currentLegIndex: Int,
+    ) = object : RouteRefreshCallback {
 
         override fun onRefresh(directionsRoute: DirectionsRoute) {
             logger.i(TAG, msg = Message("Successful route refresh"))
+            routeDiffProvider.buildRouteDiffs(
+                oldDirectionsRoute,
+                directionsRoute,
+                currentLegIndex,
+            ).forEach { logger.i(TAG, msg = Message(it)) }
```

Comment:
```suggestion
            val diffs = routeDiffProvider.buildRouteDiffs(
                oldDirectionsRoute,
                directionsRoute,
                currentLegIndex,
            )
            if (diffs.isEmpty()) {
              logger.i(TAG, msg = Message("No changes to route annotations"))
            } else {
              diffs.forEach { logger.i(TAG, msg = Message(it)) }
            }
```
Let's add something along those lines to make it clear when the response succeeds but there was no difference in data.